client/dfclient/src/gameplay_state.cpp

```diff
@@ -92,6 +92,32 @@ void GameplayState::LoadLevel() {
 		spritemap.insert(std::pair<uint16_t, std::string>(fb_Ti->gid(), fb_Ti->name()->str()));
 	}
 
+	// initialize tile layer
+	if (level->tilelayer() && level->tilelayer()->tiledata()) {
+		auto width = level->tilelayer()->width();
+		auto height = level->tilelayer()->height();
+		auto tilewidth = level->tilelayer()->tilesize()->x();
+		auto tileheight = level->tilelayer()->tilesize()->y();
+		
+		std::vector<uint16_t> tiles;
+		for(auto tile : *level->tilelayer()->tiledata()) {
+			tiles.push_back(tile);
+		}
```

Comment:
i think that creating this vector here is unnecessary, maybe you could use the tile data directly in the loops below

Reply:
Indeed. Done so